src/components/Layout/RightPaneLayout/RightPaneLayout.tsx

```diff
@@ -0,0 +1,51 @@
+import { AnimatePresence, motion } from "framer-motion";
+import React, { ReactElement } from "react";
+import { Css, Palette, px } from "../../../Css";
+import { useRightPaneContext } from "./RightPaneContext";
+
+export function RightPaneLayout({
+  children,
+  paneBgColor = Palette.White,
+  paneWidth = 450,
+}: {
+  children: ReactElement;
+  paneBgColor?: Palette;
+  paneWidth?: number;
+}) {
+  const { isRightPaneOpen, rightPaneContent, clearPane } = useRightPaneContext();
+
+  return (
+    <div css={Css.h100.df.overflowXHidden.$}>
+      <>
+        <div
+          css={
+            Css.add("width", `calc(100% - ${paneWidth + 24}px)`)
+              .add("transition", "width .2s linear")
+              .h100.mr3.overflowXAuto.if(!isRightPaneOpen).w100.mr0.$
+          }
+        >
+          {children}
+        </div>
+        <AnimatePresence>
+          {isRightPaneOpen && (
+            <motion.div
+              layout="position"
+              key="rightPane"
+              data-testid="rightPaneContent"
+              css={Css.bgColor(paneBgColor).h100.maxw(px(paneWidth)).w100.df.fdc.relative.$}
```

Comment:
**question:** I'm wondering if we should dictate that the right pane div is a `flex` parent? Doesn't seem necessary to me to define that here, but curious if there was a reason?

Reply:
Also, I am curious about the `maxw(px(paneWidth)).w100`. Could we  just define this to be `wPx(paneWidth)`? Seems like we'd be accomplishing the same thing, right? I feel like I might be missing something 🤔 . 

Ahh... Thinking through when to use the `max-width` + `width: 100%` combination,.. seems like it'd be when the element could be shown in a viewport that is thinner than the defined `paneWidth`. That way we say "take up all the space you can, but stop at 100% of the viewport to not cause a horizontal scrollbar". It's a good little trick for mobile screens, but not sure if it's a concern for this layout. Either way, doesn't need to be changed - the styles just stuck out to me.

Reply:
yeah admittedly this was brought over from the blueprint implementation but thanks for the eyes